resolver/resolver.cc

```diff
@@ -1502,6 +1507,10 @@ class ResolveSignaturesWalk {
                                            core::Names::suggestType(), move(rhs));
                 data->resultType = core::Types::untyped(ctx, sym);
             }
+        } else {
+            // we might have already resolved this constant but we want to make sure to still report some errors if
```

Comment:
can we just remove the `&& data->resultType == nullptr` instead?

Reply:
That ends up causing other problems, and in particular it "un-does" work that was already done on this constant, possibly setting `data->resultType` to be `T.untyped` when we had determined that it was already something else. This runs in the `postTransformAssign` in the resolve signatures walk, which means that plenty of work has already been done: we don't want to undo that work, but we do want to perform this (largely syntactic) error check.